includes/api/class-wc-api-taxes.php

```diff
@@ -0,0 +1,495 @@
+	 * GET /taxes/<id>
+	 *
+	 * @since 2.1
+	 * @param array $routes
+	 * @return array
+	 */
+	public function register_routes( $routes ) {
+
+		# GET/POST /taxes
+		$routes[ $this->base ] = array(
+			array( array( $this, 'get_taxes' ), WC_API_Server::READABLE ),
+			array( array( $this, 'create_tax' ), WC_API_Server::CREATABLE | WC_API_Server::ACCEPT_DATA ),
+		);
+
+		# GET /taxes/count
+		$routes[ $this->base . '/count'] = array(
+			array( array( $this, 'get_taxes_count' ), WC_API_Server::READABLE ),
+		);
+
+		# GET/PUT/DELETE /taxes/<id>
+		$routes[ $this->base . '/(?P<id>\d+)' ] = array(
+			array( array( $this, 'get_tax' ), WC_API_Server::READABLE ),
+			array( array( $this, 'edit_tax' ), WC_API_SERVER::EDITABLE | WC_API_SERVER::ACCEPT_DATA ),
+			array( array( $this, 'delete_tax' ), WC_API_SERVER::DELETABLE ),
+		);
+
+		# GET/POST /taxes/classes
+		$routes[ $this->base . '/classes' ] = array(
+			array( array( $this, 'get_tax_classes' ), WC_API_Server::READABLE ),
+			array( array( $this, 'create_tax_class' ), WC_API_Server::CREATABLE | WC_API_Server::ACCEPT_DATA ),
+		);
+
+		# GET /taxes/classes/count
+		$routes[ $this->base . '/classes/count'] = array(
+			array( array( $this, 'get_tax_classes_count' ), WC_API_Server::READABLE ),
+		);
+
+		# GET /taxes/classes/<slug>
+		$routes[ $this->base . '/classes/(?P<slug>\w[\w\s\-]*)' ] = array(
+			array( array( $this, 'delete_tax_class' ), WC_API_SERVER::DELETABLE ),
+		);
+
+		# POST|PUT /taxes/bulk
+		$routes[ $this->base . '/bulk' ] = array(
+			array( array( $this, 'bulk' ), WC_API_Server::EDITABLE | WC_API_Server::ACCEPT_DATA ),
+		);
+
+		return $routes;
+	}
+
+	/**
+	 * Get all taxes
+	 *
+	 * @since 2.5.0
+	 *
+	 * @param string $fields
+	 * @param array  $filter
+	 * @param string $class
+	 * @param int    $page
+	 *
+	 * @return array
+	 */
+	public function get_taxes( $fields = null, $filter = array(), $class = null, $page = 1 ) {
+		if ( ! empty( $class ) ) {
+			$filter['tax_rate_class'] = $class;
+		}
+
+		$filter['page'] = $page;
+
+		$query = $this->query_tax_rates( $filter );
+
+		$taxes = array();
+
+		foreach ( $query['results'] as $tax ) {
+			$taxes[] = current( $this->get_tax( $tax->tax_rate_id, $fields ) );
+		}
+
+		// Set pagination headers
+		$this->server->add_pagination_headers( $query['headers'] );
+
+		return array( 'taxes' => $taxes );
+	}
+
+	/**
+	 * Get the tax for the given ID
+	 *
+	 * @since 2.5.0
+	 *
+	 * @param int $id The tax ID
+	 * @param string $fields fields to include in response
+	 *
+	 * @return array|WP_Error
+	 */
+	public function get_tax( $id, $fields = null ) {
+		global $wpdb;
+
+		try {
+			$id = absint( $id );
+
+			// Permissions check
+			if ( ! current_user_can( 'manage_woocommerce' ) ) {
+				throw new WC_API_Exception( 'woocommerce_api_user_cannot_read_tax_rate', __( 'You do not have permission to read tax rate', 'woocommerce' ), 401 );
+			}
+
+			// Get tax rate details
+			$tax = $wpdb->get_row( $wpdb->prepare( "
```

Comment:
might be worth abstracting the retrieval into it's own global function so it's usable by non-API code as well